src/request/export/ExportWords.js

```diff
@@ -21,94 +22,112 @@ class ExportWords extends Nimiq.Observable {
-        this._privacyWarningPassphraseBox = new PassphraseBox(
-            $privacyWarningPassphraseBox, {
-                buttonI18nTag: 'passphrasebox-continue',
-                hideInput: !this._request.keyInfo.encrypted,
-                minLength: this._request.keyInfo.hasPin ? 6 : undefined,
-                hideCancel: true,
-            },
-        );
+        // components
+        this._wordsPasswordBox = new PassphraseBox($wordsPasswordBox, {
+            buttonI18nTag: 'passphrasebox-show-words',
+            hideInput: !request.keyInfo.encrypted || !!this._key,
+            hideCancel: true,
+        });
         this._recoveryWords = new RecoveryWords($recoveryWords, false);
         this._validateWords = new ValidateWords($validateWords);
-
-        $privacyWarningButton.addEventListener('click', this._goToShowWords.bind(this));
-        $recoveryWordsButton.addEventListener('click', this._goToValidateWords.bind(this));
-        this._privacyWarningPassphraseBox.on(PassphraseBox.Events.SUBMIT, this._passphraseSubmitted.bind(this));
-        this._validateWords.on(ValidateWords.Events.VALIDATED, this._finish.bind(this));
-        this._validateWords.on(ValidateWords.Events.BACK, this._goToShowWords.bind(this));
-        this._validateWords.on(ValidateWords.Events.SKIP, this._finish.bind(this));
+        /* eslint-disable no-new */
+        new ProgressIndicator(this._$noRecovery.querySelector('.progress-indicator'), 3, 1);
+        new ProgressIndicator($recoveryWordsPage.querySelector('.progress-indicator'), 3, 2);
+        new ProgressIndicator($validateWordsPage.querySelector('.progress-indicator'), 3, 3);
+        /* eslint-enable no-new */
+
+        // events
+        $noRecoverySkip.addEventListener('click', event => {
+            event.preventDefault();
+            this._resolve({ success: false });
+        });
+        $recoveryWordsSkip.addEventListener('click', event => {
+            event.preventDefault();
+            this._resolve({ success: true });
+        });
+        this._wordsPasswordBox.on(PassphraseBox.Events.SUBMIT, this._passphraseSubmitted.bind(this));
+        $recoveryWordsContinue.addEventListener('click', () => {
+            this._validateWords.reset();
+            window.location.hash = ExportWords.Pages.VALIDATE_WORDS;
+        });
+        this._validateWords.on(ValidateWords.Events.VALIDATED, () => this._resolve({ success: true }));
+        this._validateWords.on(ValidateWords.Events.SKIP, () => this._resolve({ success: true }));
     }
 
     run() {
-        this._privacyWarningPassphraseBox.reset();
-        window.location.hash = ExportWords.Pages.PRIVACY;
+        this._wordsPasswordBox.reset();
+        window.location.hash = ExportWords.Pages.RECOVERY_WORDS_INTRO;
         if (TopLevelApi.getDocumentWidth() > Constants.MIN_WIDTH_FOR_AUTOFOCUS) {
-            this._privacyWarningPassphraseBox.focus();
+            this._wordsPasswordBox.focus();
```

Comment:
I guess this could lead to the keyboard opening on e.g. an iPad, which fullfills the min-width condition, when the input is however hidden because it's unencrypted or the key is already known. So I think this focus should be only executed when we know we need a password.

Or, better: The PassphraseBox should only forward the focus() to the input element, when its `options.hideInput` is false.